src/furax/interfaces/toast/mapmaker/interface.py

```diff
@@ -0,0 +1,108 @@
+from dataclasses import dataclass
+from functools import cached_property
+
+import jax.numpy as jnp
+import numpy as np
+import toast
+from astropy import units as u
+from jaxtyping import Array, Float, Int
+from toast.observation import default_values as defaults
+
+from .utils import get_local_meridian_angle
+
+
+@dataclass
+class ObservationData:
```

Comment:
I am assuming this is some static data class that it used internally
however it is still very easy to make into an actual jittable pytree in case it is passed around 

```suggestion

@jax.tree_util.register_dataclass
@dataclass
class ObservationData:
```
check
https://jax.readthedocs.io/en/latest/_autosummary/jax.tree_util.register_dataclass.html

Reply:
Thank you! I simply added this line for now. In the future, I will look into specifying meta/data fields for performance.